File: javascript/lasagna-master/lasagna-master.js
/// <reference path="./global.d.ts" />

import { allowedNodeEnvironmentFlags } from "process";

// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(timeOnTimer) {
  if (timeOnTimer === 0) {
    return "Lasagna is done.";
  } else if (timeOnTimer > 0) {
    return "Not done, please wait.";
  } else {
    return "You forgot to set the timer.";
  }
}

export function preparationTime(layers, avgPrepTime = 2) {
  return layers.length * avgPrepTime;
}

export function quantities(layers) {
  let noodles = 0;
  let sauce = 0;

  for (const layer of layers) {
    if (layer === "noodles") {
      noodles += 50;
    } else if (layer === "sauce") {
      sauce += 0.2;
    }
  }

  return {
    noodles,
    sauce,
  };
}

export function addSecretIngredient(friendList, ownList) {
  ownList.push(friendList[friendList.length - 1]);
}

export function scaleRecipe(recipe, scale) {
  let scaledRecipe = {};

  for (const layer in recipe) {
    scaledRecipe[layer] = recipe[layer] * (scale / 2);
  }

  return scaledRecipe;
}

File: go/lasagna-master/lasagna_master.go
package lasagna

func PreparationTime(layers []string, avgPrepTime int) int {
	if avgPrepTime <= 0 {
		avgPrepTime = 2
	}
	return len(layers) * avgPrepTime
}

func Quantities(layers []string) (int, float64) {
	noodles := 0
	sauce := 0.0
	for _, layer := range layers {
		if layer == "noodles" {
			noodles++
		} else if layer == "sauce" {
			sauce++
		}
	}
	return noodles * 50, float64(sauce) * 0.2
}

func AddSecretIngredient(friendList []string, ownList []string) {
	ownList[len(ownList)-1] = friendList[len(friendList)-1]
}

func ScaleRecipe(ingredients []float64, portions int) []float64 {
	var scaledRecipe []float64
	for _, ingredient := range ingredients {
		scaledRecipe = append(scaledRecipe, ingredient/2*float64(portions))
	}
	return scaledRecipe
}
